core/src/main/java/jenkins/model/IdStrategy.java

```diff
@@ -207,9 +209,14 @@ public String keyFor(@NonNull String id) {
             return id.toLowerCase(Locale.ENGLISH);
         }
 
+        @Override
+        public boolean equals(@NonNull String id1, @NonNull String id2) {
+            return id1.equalsIgnoreCase(id2);
```

Comment:
given in AUthZ we only care that a user has the role we do not want to go matching individual characters when there are many groups all with similar prefixes 

for example consider `DG_DEPT_ENGINEERING_JENKINS_ADMIN` and  `DG_DEPT_ENGINEERING_JENKINS_USER`

a user with the later and a ACL granting a permission on the former we would have ended up checking the first 29 characters before deciding that the user did not match, when the string length would have told us that pretty cheaply.
combine this with lots of groups and the performance starts to take a hit.